### DISCUSSION TITLE
Function parameter Assertions

### DISCUSSION THREAD

**Discussion by philipszalla** [dotnet/csharplang#5234] [Language Ideas]
Would it be possible to introduce an `assert` keyword, which gets a condition and throws an `ArgumentException`? Similar to the Java assert.

example:
```csharp
public string Greet(string name)
{
  assert !string.IsNullOrEmpty(name);

  return "Hello " + name;
}
```

instead of
```csharp
public string Greet(string name)
{
  if (string.IsNullOrEmpty(name))
    throw new ArgumentException(nameof(name));

  return "Hello " + name;
}
```

**Comment by svick** [comment]
Why does this need to be a language keyword? Wouldn't a function be enough?

One thing that would be harder to do using a function would be getting the name of the argument without having to repeat it. But that might be doable using C# 10.0 caller argument expression (https://github.com/dotnet/csharplang/issues/287).
